plugins/config/parsers/auto_constructed_tables.cpp

```diff
@@ -105,11 +105,11 @@ Status ATCConfigParserPlugin::setUp() {
 Status ATCConfigParserPlugin::update(const std::string& source,
                                      const ParserConfig& config) {
   auto cv = config.find(kParserKey);
-  if (cv == config.end()) {
-    return Status(1, "No configuration for ATC (Auto Table Construction)");
+  if (cv == config.end() || !cv->second.doc().IsObject()) {
+    return Status::success();
```

Comment:
I'm confused why this is returning a `Status::success()`. If the object wasn't found, or `.IsObject()` was false shouldn't this be a `::failure`?

Reply:
That's a good point. Let me re-check the parsers code and see what the convention is.

Reply:
Returning a success when the input is not the right JSON-type is the most common handling. I created https://github.com/osquery/osquery/issues/6037 to do a deep-dive into what is the ideal handling and how to make this more abstract/easy.